enhancements/network/on-prem-mutable-vips.md

```diff
@@ -0,0 +1,199 @@
+
+* Configuration of any VIPs beyond a second pair for the second IP stack. MetalLB is a better solution for creating arbitrary loadbalancers.
+
+## Proposal
+
+### Workflow Description
+
+The proposed worklflow after implementing the feature would look as described below
+
+1. Administrator of a dual-stack cluster with single-stack VIPs wants to add a second pair for the second IP stack configured.
+
+1. Administrator edits the Infrastructure CR named `cluster` by changing the `spec.platformSpec.[*].apiServerInternalIPs` and `spec.platformSpec.[*].ingressIPs` fields.
+
+1. Cluster Network Operator picks the modification of the object and compares values with `spec.platformStatus.[*].apiServerInternalIPs` and `spec.platformStatus.[*].ingressIPs`.
+
+1. After validating that the requested change is valid (i.e. conforms with the goals and non-goals), the change is propagated down to the keepalived template and configuration file.
+
+1. After keepalived configuration is changed, the service is restarted or reloaded to apply the changes.
+
+### API Extensions
+
+New fields will have to be added to the platform spec section for [baremetal](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L694), [vsphere](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L1089) and [OpenStack](https://github.com/openshift/api/blob/938af62eda38e539488d6193e7af292e47a09a5e/config/v1/types_infrastructure.go#L772). In the first implementation only baremetal will be covered.
+
+### Implementation Details/Notes/Constraints
+
+Because of the expertise and the team owning this feature, we think the best place for implementing the logic is Cluster Network Operator. CNO however is not an owner of the `cluster` Infrastructure CR what at first sight may seem like a challenge.
+
+Kubernetes API provide a solution for this and we can leverage Watches (with Predicate for optimization) to allow a controller inside CNO to watch an object it does not own. This in fact already happens inside CNO in the [infrastructureconfig controller](https://github.com/openshift/cluster-network-operator/blob/2005bcd8c93de5bffc05c9c943b51386007f6b9a/pkg/controller/infrastructureconfig/infrastructureconfig_controller.go#L47).
+
+The controller would get a logic implemented that validates if the change requested by user is valid as in the first implementation we only want to allow adding second pair of VIPs but forbid any modifications of the already configured ones.
+
+The current values of `spec.platformStatus.[*].xxx` are propagated during installation by XXX. After implementing this feature, the same XXX would also set `spec.platformSpec.[*].xxx` so that the cluster is installed with a correct configuration.
```

Comment:
I believe platformStatus is populated by the installer. Like here: https://github.com/openshift/installer/blob/186cb916c388d29ed3f6ef4e71d9fda409f30bdf/pkg/asset/manifests/infrastructure.go#L160

Reply:
Yep, seems right. Updated the doc